### PR TITLE
feat(sbt): enhanced extract capabilities

### PR DESCRIPTION
**Pull request by darl**
1. Its common pattern to define sbt dependencies in separate scala file inside `project` folder:
https://github.com/search?q=filename%3Adependencies.scala&type=Code

2. parsing dependency declaration inside variable:
```scala
lazy val ujson = "com.lihaoyi" %% "ujson" % "0.7.1"
```

3. normalization of scala version (2.12.10 -> 2.12)

4. repo.maven.org and other repositories expect trailing slash
or internal artifactory installation respond with 404 status on requests without trailing slash

### CODE REVIEW

**Comment by rarkins** [comment]
Great! Can you add some unit tests that would fail on the current master but pass with the new code? I want to reduce the chances we ever regress on these capabilities.

**Comment by darl** [comment]
@rarkins Done.
Can you help with semantic title?

**Comment by rarkins** [comment]
@darl I can, but I'm also thinking of splitting this into separate commits to make it clearer for future

File: lib/datasource/sbt/index.ts [hunk 1]
```diff
@@ -9,7 +9,7 @@ async function resolvePackageReleases(
   artifact: string,
   scalaVersion: string
 ): Promise<string[]> {
-  const indexContent = await downloadHttpProtocol(searchRoot, 'sbt');
+  const indexContent = await downloadHttpProtocol(searchRoot + '/', 'sbt');
```
Comment (rarkins) [inline lib/datasource/sbt/index.ts]: Is there any concern here of what happens if the `searchRoot` already had a trailing slash?

File: lib/datasource/sbt/index.ts [hunk 2]
```diff
@@ -31,7 +31,7 @@ async function resolvePackageReleases(
       parseIndexDir(content, x => !/^\.+$/.test(x));
     for (const searchSubdir of searchSubdirs) {
       const content = await downloadHttpProtocol(
-        `${searchRoot}/${searchSubdir}`,
+        `${searchRoot}/${searchSubdir}/`,
```
Comment (rarkins) [inline lib/datasource/sbt/index.ts]: Are subdirs also always without a trailing slash?

File: lib/manager/sbt/extract.ts [hunk 3]
```diff
@@ -20,6 +20,11 @@ const isScalaVersion = (str: string): boolean =>
 const getScalaVersion = (str: string): string =>
   str.replace(/^\s*scalaVersion\s*:=\s*"/, '').replace(/"\s*$/, '');
 
+const normalizeScalaVersion = (str: string): string =>
```
Comment (rarkins) [inline lib/manager/sbt/extract.ts]: Can the reasoning for this be documented with a code comment and/or with a reference to documentation somewhere?

**Review by rarkins — CHANGES_REQUESTED** [review]
If there's a chance that urls may already have a trailing slash, it would be safer to do something like `existing.replace(/\/?$/, '/')` instead of `existing + '/'`. Or perhaps make that into a `ensureTrailingSlash()` function so that it's not duplicated everywhere.

**Review by zharinov — DISMISSED** [review]
Looks good to me

File: lib/manager/sbt/extract.ts (hunk 3 above)
Comment (darl) [reply lib/manager/sbt/extract.ts]: Done

**Comment by darl** [comment]
added method `ensureTrailingSlash` 
and comment with link to scala artifact naming convention

**Comment by rarkins** [comment]
I've separated out the two fixes and now merged back into this branch. Will merge to master once tests pass

**Comment by rarkins** [comment]
Thanks, @darl!

**Pull request merged**